packages/firestore/src/core/firestore_client.ts

```diff
@@ -536,7 +536,6 @@ export class FirestoreClient {
   }
 
   unlisten(listener: QueryListener): void {
-    this.verifyNotShutdown();
     this.asyncQueue.enqueueAndForget(() => {
```

Comment:
This potentially calls into the SyncEngine. I wonder if we should guard this on the shutdown state and not call `unlisten` if we are shut down?

Reply:
Done.